shm/Ship.cpp

```diff
@@ -41,20 +42,37 @@ Ship& Ship::operator+=(const size_t crew) {
 }
 
 void Ship::load(const std::shared_ptr<Cargo>& cargo) {
-    cargo_.push_back(cargo);
+    auto it = std::find_if(begin(cargo_), end(cargo_),
+                           [&cargo](const auto& ptr) {
+                               return *cargo == *ptr;
+                           });
+
+    if (it != end(cargo_)) {
+        *(*it) += cargo->getAmount();
+    } else {
+        cargo_.push_back(cargo);
+    }
```

Comment:
You can take a look at my branch https://github.com/niemarcin/object-oriented-programming/blob/crazyShopping/shm/Ship.cpp

I have there additional `findGivenCargoInStock()` private method - I think it might be more clear to read it.

But, I have little bit different implementation of `load()` as I also use the `amount` parameter.

Reply:
will replace it, I forgot I left this that ugly